### ISSUE TITLE
[BSI][20200601] Sensitive Data Stored in Unencrypted Database on Filesystem

### ISSUE THREAD

**Issue by BSI-TF-CWA** [corona-warn-app/cwa-app-ios#125]
Rating: Low
 
Description:
The iOS app stores sensitive data in an unencrypted sqlite Database in the
app's sandbox. An attacker with root privileges, or code execution in the
scope of the CWA app can read this information. This includes data, such as
the TeleTAN or registration token. A code comment mentions that this database
"still needs to be encrypted" [1]. If this measure is implemented, the issue
is mitigated.

However, it is questionable why the TeleTAN needs to be stored at all. As it
should only be used once to create a ``sessionToken``, there should be no need
to store the TeleTAN on disk.

[1]: cwa-app-ios/src/xcode/ENA/ENA/Source/Services/ExposureSubmissionService.swift:286

Proof of Concept:
An excerpt of the binary content of the ``secureStore.sqlite`` in ``Library/Application Support``:
```
secureStore.sqlite
SQLite format 3
{tablekvkv
CREATE TABLE kv (
    key TEXT UNIQUE,
    value BLOB
indexsqlite_autoindex_kv_1kv
teleTan"[REDACTED]"
isOnboardedtrue0
[...]
```

**Comment by inf2381** [comment]
Thanks for the reminder 😊 This will be fixed in #85.

**Comment by ChristianKienle** [comment]
@BSI-TF-CWA We have just merged #85 which added SQLCipher to the project.

**Comment by thomasaugsten** [comment]
Teletan store is removed as well.
